File: Day-02/python/0skxr/part1.py
id, value, f = 1, 0, open("/input.txt", "r")
for line in f.readlines():
  sets, isInBounds = line.split(":")[1:len(line)][0].split(";"), True
  for set in sets: r, g, b, currentSet = 0, 0, 0, set.split(",")
    for sample in currentSet:
      if (sample.strip().endswith("red")): r += int(sample.strip().strip(" red"))
      elif (sample.strip().endswith("green")): g += int(sample.strip().strip(" green"))
      elif (sample.strip().endswith("blue")): b += int(sample.strip().strip(" blue"))
    if (not (r <= 12 and g <= 13 and b <= 14)): isInBounds = False
  if (isInBounds): value += id 
  id += 1
print(value)

File: Day-02/javascript/HowToCodeDE/part1.js
const fs = require("fs");
const data = fs.readFileSync("input.txt").toString();
const limit = {
    "red": 12,
    "green": 13,
    "blue": 14,
};
let gamecount = 0;
const games = data.split(/\r?\n|\r|\n/g);
games.forEach(game => {
    let gameImpossible = 0;
    let gameId = parseInt(game.split(":")[0].replace("Game ", ""));
    let values = game.split(": ")[1];
    let runs = values.split("; ");
    runs.forEach(run => {
        let cubes = run.split(", ");
        cubes.forEach(cube => {
            let count = cube.split(" ");
            if ( parseInt(count[0]) > limit[count[1]]) gameImpossible = 1;
        });
    });
    if (gameImpossible == 0) {
        gamecount += gameId;
    }
});
console.log(gamecount);
